gensim/models/keyedvectors.py

```diff
@@ -985,10 +989,14 @@ def most_similar_cosmul(self, positive=None, negative=None, topn=10):
 
         self.fill_norms()
 
-        if isinstance(positive, str) and not negative:
-            # allow calls like most_similar_cosmul('dog'), as a shorthand for most_similar_cosmul(['dog'])
+        if isinstance(positive, KEY_TYPES + (ndarray,)):
+            # allow passing a single string-key or vector for the positive argument
             positive = [positive]
 
+        if isinstance(negative, KEY_TYPES + (ndarray,)):
```

Comment:
You're duplicating the key types manipulation in 4 places.

Please do something like:

```
key_types = KEY_TYPES + (ndarray, )
if instance(positive, key_types):
    ...
```